File: programming-essentials/pe-05/practice.py
score_text = "80"
project_completed_text = "true"
preferred_field_text = "backend"

try:
    score = float(score_text)
    project_completed = project_completed_text == "true"
    preferred_field = preferred_field_text.strip().lower()

    valid_fields = ["frontend", "backend", "data"]

    is_valid = (
        0 <= score <= 100
        and preferred_field in valid_fields
    )

    if not is_valid:
        print("Invalid input.")

    elif score >= 80 and project_completed:
        print("Next step: Advanced project.")
        print(
            "Rule: Score is at least 80 "
            "and the project is completed."
        )

    elif score < 60 or not project_completed:
        print("Next step: Review and practice.")
        print(
            "Rule: Score is below 60 "
            "or the project is not completed."
        )

    else:
        print(
            f"Next step: Continue "
            f"{preferred_field} learning."
        )
        print(
            "Rule: The trainee is developing "
            "and can continue learning."
        )

except ValueError:
    print("Invalid score.")

File: programming-essentials/pe-05/practice.ts
const scoreText: string = "80";
const projectCompletedText: string = "true";
const preferredFieldText: string = "backend";

const score: number = Number(scoreText);
const projectCompleted: boolean = projectCompletedText === "true";
const preferredField: string = preferredFieldText.trim().toLowerCase();

const validFields: string[] = ["frontend", "backend", "data"];

const isValid: boolean =
    Number.isFinite(score) &&
    score >= 0 &&
    score <= 100 &&
    validFields.includes(preferredField);

if (!isValid) {
    console.log("Invalid input.");
} else if (score >= 80 && projectCompleted) {
    console.log("Next step: Advanced project.");
    console.log("Rule: Score is at least 80 and the project is completed.");
} else if (score < 60 || !projectCompleted) {
    console.log("Next step: Review and practice.");
    console.log("Rule: Score is below 60 or the project is not completed.");
} else {
    console.log(`Next step: Continue ${preferredField} learning.`);
    console.log("Rule: The trainee is developing and can continue learning.");
}

export {};